public/v4/apps/affine.yml

```diff
@@ -1,143 +1,165 @@
-            ]
-        ports:
-            - '5555:5555'
+        image: ghcr.io/toeverything/affine:$$cap_affine_version
+        # No command is needed, uses the image's default CMD
         depends_on:
             $$cap_appname-redis:
                 condition: service_healthy
             $$cap_appname-postgres:
                 condition: service_healthy
+            $$cap_appname-migration:
+                condition: service_completed_successfully
         volumes:
-            # custom configurations
-            - $$cap_appname-db-data:/root/.affine/config
-            # blob storage
-            - $$cap_appname-db-data:/root/.affine/storage
+            - $$cap_appname-config:/root/.affine/config
+            - $$cap_appname-storage:/root/.affine/storage
         logging:
             driver: 'json-file'
             options:
                 max-size: '1000m'
         restart: unless-stopped
         environment:
-            NODE_OPTIONS: '--import=./scripts/register.js'
-            AFFINE_CONFIG_PATH: /root/.affine/config
             REDIS_SERVER_HOST: srv-captain--$$cap_appname-redis
             DATABASE_URL: postgres://affine:$$cap_db_affine_password@srv-captain--$$cap_appname-postgres:5432/affine
             NODE_ENV: production
-            AFFINE_ADMIN_EMAIL: $$cap_affine_admin_email
-            AFFINE_ADMIN_PASSWORD: $$cap_affine_admin_password
+            AFFINE_INDEXER_ENABLED: 'false'
+            # CapRover variables for setup
             AFFINE_SERVER_HOST: $$cap_affine_server_host
             AFFINE_SERVER_HTTPS: $$cap_affine_server_https
             AFFINE_SERVER_EXTERNAL_URL: $$cap_affine_server_external_url
+            # Mailer settings
             MAILER_HOST: $$cap_mailer_host
             MAILER_PORT: $$cap_mailer_port
             MAILER_USER: $$cap_mailer_user
             MAILER_PASSWORD: $$cap_mailer_password
             MAILER_SENDER: $$cap_mailer_sender
-            # Telemetry allows us to collect data on how you use the affine. This data will helps us improve the app and provide better features.
-            # Uncomment next line if you wish to quit telemetry.
-            # - TELEMETRY_ENABLE=false
+            # Telemetry
+            # TELEMETRY_ENABLE: 'false'
         caproverExtra:
             containerHttpPort: '3010'
             websocketSupport: true
+
+    # Affine Database Migration Job
+    $$cap_appname-migration:
+        image: ghcr.io/toeverything/affine:$$cap_affine_version
+        command: ['sh', '-c', 'node ./scripts/self-host-predeploy.js']
+        depends_on:
+            $$cap_appname-redis:
+                condition: service_healthy
+            $$cap_appname-postgres:
+                condition: service_healthy
+        volumes:
+            - $$cap_appname-config:/root/.affine/config
+            - $$cap_appname-storage:/root/.affine/storage
+        restart: on-failure # Run as a one-off job
```

Comment:
This is ignored in CapRover. What would we the downside of running this migration potentially more than once? Is it smart enough to do nothing the second time it runs?

If so, you can change it to:
```
command: ["sh", "-c", "node ./scripts/self-host-predeploy.js && tail -f /dev/null"]
```

Also change the image name to `$$cap_appname-migration-delete-me-after-install`

Reply:
Also, is this really required even for a fresh install?